sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -2085,4 +2087,28 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
       }
     }
   }
+
+  test("SPARK-28431: prevent CSV datasource throw TextParsingException with large size message") {
+    withTempDir { dir =>
+      val maxCharsPerCol = 10000
+      val str = "a" * (maxCharsPerCol + 1)
+
+      val csvFile = new File(dir, "data.csv")
+      Files.write(
+        csvFile.toPath,
+        Seq(str).asJava,
```

Comment:
nit: `str.getBytes()`?

Reply:
I don't think so, here I need to write a text file, not a binary file.

Reply:
I think @MaxGekk just meant a nit: write a file with explicitly calling `getBytes()`.

I think it's better to call `getBytes(StandardCharset.UTF-8)` actually. Otherwise, the codes look like it's going to write out unicodes as are.

Reply:
and .. CSV datasource are pretty sensitive about encodings - I and @MaxGekk had some extreme discussions about that before ... It might be better to write out UTF-8 just to be more explicit :-) .